src/datafinder/gui/user/dialogs/connect_dialog.py

```diff
@@ -86,12 +86,13 @@ def __init__(self, parent=None, preferences=None):
         self.connect(self.preferencesButton, QtCore.SIGNAL("clicked()"), self._preferencesActionSlot)
         self.uri = preferences.connectionUris
         connection = self._preferences.getConnection(self._getUrl())
-        self._useLdap = connection.useLdap
-        self._ldapBaseDn = connection.ldapBaseDn
-        self._ldapServerUri = connection.ldapServerUri
-        self._useLucene = connection.useLucene
-        self._luceneIndexUri = connection.luceneIndexUri
-                     
+        if connection is not None:

```

Comment:
The only problem I see is that the properties are no longer defined in this case. I.e., if we have methods which reads one of them we end up with an AttributeError... right?